gradle/libs.versions.toml

```diff
@@ -0,0 +1,153 @@
+# Also see settings.gradle file for dynamic versionCatalogs dependencies
+
+[versions]
+antlr = "4.9.3"
+deltaStandalone = "0.6.0"
+deltaCore = "2.2.0"
+flink115 = { strictly = "[1.15, 1.16[", prefer = "1.15.0"}
+flink116 = { strictly = "[1.16, 1.17[", prefer = "1.16.2"}
+flink117 = { strictly = "[1.17, 1.18[", prefer = "1.17.1"}
```

Comment:
I think we should be using a very specific flink version for 1.15 / 1.16 / 1.17 as I believe there have been breaking changes in the past between patch versions

Reply:
For places like this(where a rich version declaration with  strictly & prefer is used) and similar others I will put a single comments: please read a head description to PR, I've updated it with a point about rich versions. Let me know if you have questions and/or suggestions.

Reply:
so this is only done to (potentially) make dependabot work? I think we should first go with the normal version definition for Flink + Jackson + Tez + Hive.
If we run into issues with Dependabot, then we'll re-evalute but to me this version definition indicates that we might end up using a different patch version of a particular library than defined here, which could be difficult to debug

Reply:
So it seems dependabot indeed needs the rich version as can be seen in https://github.com/nastra/iceberg/pull/98, as otherwise dependabot will try to align all versions. I think we should add a comment to the rich version definition why it is being done

Reply:
done
added a top comments and references to it